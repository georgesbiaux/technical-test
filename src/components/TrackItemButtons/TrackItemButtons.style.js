import styled from 'styled-components';

const StyledPriorityButton = styled.div`
  background-color: #fbfbfb;
`;

export default StyledPriorityButton;
