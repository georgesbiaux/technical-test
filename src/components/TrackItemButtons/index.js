export { default } from './TrackItemButtons.container';
