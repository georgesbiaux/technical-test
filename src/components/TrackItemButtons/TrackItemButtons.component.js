import React from 'react';
import { Button, VoteButton } from 'components';
import StyledTrackItemButtons from './TrackItemButtons.style';

const TrackItemButtons = props => (
  <StyledTrackItemButtons className="track-item-button">
    <Button
      icon="fa-bars"
      priority={props.track.priority}
      onClick={() => props.toggleTrackPriority(props.track.id)}
    />
    <VoteButton
      votes={props.track.votes.count}
      userVoted={props.track.votes.userVoted}
      onClick={() => props.toggleUserVote(props.track.id)}
    />
    <Button
      icon="fa-trash"
      priority={props.track.priority}
      onClick={() => props.removeTrack(props.track.id)}
    />
  </StyledTrackItemButtons>
);

export default TrackItemButtons;
