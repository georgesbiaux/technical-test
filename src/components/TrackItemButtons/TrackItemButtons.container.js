import { connect } from 'react-redux';
import {
  toggleTrackPriority,
  toggleUserVote,
  removeTrack,
} from 'redux/TrackList/actions';
import { toJS } from '../../services/immutable/toJs';
import TrackItemButtons from '../TrackItemButtons/TrackItemButtons.component';

function mapStateToProps() {
  return {};
}

function mapDispatchToProps(dispatch) {
  return {
    toggleTrackPriority: trackId => dispatch(toggleTrackPriority(trackId)),
    toggleUserVote: trackId => dispatch(toggleUserVote(trackId)),
    removeTrack: trackId => dispatch(removeTrack(trackId)),
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(
  toJS(TrackItemButtons),
);
