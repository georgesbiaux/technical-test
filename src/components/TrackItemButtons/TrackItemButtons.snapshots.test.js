import React from 'react';
import { render } from 'enzyme';
import toJSON from 'enzyme-to-json';
import TrackItemButtons from './TrackItemButtons.component';

describe('<TrackItemButtons>', () => {
  it('should render correctly', () => {
    const props = {
      track: {
        id: 1,
        name: 'Beast Of Burden',
        duration: 211,
        priority: false,
        artist: 'The Rolling Stones',
        adder: {
          id: 1,
          name: 'kant',
          pictureUrl:
            'https://res.cloudinary.com/jukeo-net/image/upload/ano-b2_eezggd',
        },
        pictureUrl:
          'https://i.scdn.co/image/adcf79a6adae4e1a44d0e34860cc103e28ee4372',
        votes: {
          count: 0,
          userVoted: false,
        },
      },
    };
    const tree = render(<TrackItemButtons {...props} />);
    expect(toJSON(tree)).toMatchSnapshot();
  });
});
