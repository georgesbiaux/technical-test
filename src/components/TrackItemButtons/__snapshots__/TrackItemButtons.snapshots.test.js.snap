// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrackItemButtons> should render correctly 1`] = `
<div
  class="track-item-button sc-gZMcBi iZyrSi"
>
  <button
    class="priority-button sc-htoDjs esxIvP"
  >
    <i
      class="fa fa-bars"
    >
      
    </i>
  </button>
  <button
    class="vote-button sc-dnqmqq kTAFid"
  >
    0
    <i
      class="fa fa-heart"
    >
      
    </i>
  </button>
</div>
`;
